app/src/main/java/com/globant/example/mentorapp/presentation/di/Module/ApplicationModule.java

```diff
@@ -0,0 +1,129 @@
+import com.globant.example.mentorapp.presentation.view.fragment.ListUsersFragment;
+import com.squareup.otto.Bus;
+import com.squareup.otto.ThreadEnforcer;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Singleton;
+
+import dagger.Module;
+import dagger.Provides;
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+/**
+ * Class to control UserList injections
+ * Created by oscar.amezquita on 8/06/2017.
+ */
+@Module
+public class ApplicationModule {
+
+    public Context context;
+    public MentorApplication application;
+
+    public ApplicationModule(MentorApplication application) {
+        this.context = application;
+        this.application = application;
+    }
+
+    @Provides
+    protected LinearLayoutManager provideLinearLayoutManager() {
+        return new LinearLayoutManager(context);
+    }
+
+    @Provides
+    @Singleton
+    public Context provideContext() {
+        return context;
+    }
+
+    @Provides
+    @Singleton
+    protected MentorApplication provideApplication() {
+        return application;
+    }
+
+    @Provides
+    @Singleton
+    protected ListUsersInteractorImpl provideUserInteractor() {
+        return new ListUsersInteractorImpl(application.getApplicationComponent());
+    }
+
+    @Provides
+    protected ListUsersFragment provideListUsersFragment() {
+        return new ListUsersFragment();
+    }
+
+    @Provides
+    @Singleton
+    ListUsersPresenterImpl provideListUsersPresenter() {
+        return new ListUsersPresenterImpl(application.getApplicationComponent());
+    }
+
+    @Provides
+    @Singleton
+    public Retrofit provideRetrofitClient(OkHttpClient client) {
+        return new Retrofit.Builder().client(client).baseUrl(ApiUtils.BASE_URL).addConverterFactory(GsonConverterFactory.create())
+                .build();
+    }
+
+    @Provides
+    @Singleton
+    public APIClient providesApiClient() {
+        return new APIClient(application.getApplicationComponent());
+    }
+
+    @Provides
+    @Singleton
+    public OkHttpClient provideOkHttpClient() {
+        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
+        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
+        OkHttpClient.Builder b = new OkHttpClient.Builder();
+        b.readTimeout(2000, TimeUnit.MILLISECONDS);
+        b.writeTimeout(4000, TimeUnit.MILLISECONDS);
+        b.addInterceptor(interceptor);
+        return b.build();
+
+    }
+
+    @Provides
+    @Singleton
+    public Bus provideBus() {
+        return new Bus(ThreadEnforcer.ANY);
+    }
+
+    @Provides
+    @Singleton
+    public MutableLiveData<List<UserEntity>> provideLiveData() {
+        return new MutableLiveData<>();
+    }
+
+    @Provides
+    @Singleton
+    DividerItemDecoration providetemDecoration() {
```

Comment:
You're generating a singleton of DividerItemDecoration. What if in some other flow you need a different divider decoration?

1.- You're creating before really using it
2.- It's better to create it dynamically.